docs/reference/specifications/providers.md

```diff
@@ -0,0 +1,251 @@
+---
+description: flagd provider specification
+---
+
+# flagd Providers
+
+!!! note
+
+    This document serves as both a specification and general documentation for flagd providers.
+    For language-specific details, see the `README.md` for the provider in question.
+
+flagd providers are as essential as the flagd daemon itself, acting as the "bridge" between the OpenFeature SDK and flagd.
+In fact, flagd providers may be the most crucial part of the flagd framework, as they can be used without an active flagd instance.
+This document outlines their behavior and configuration.
+
+## In-Process vs RPC Evaluation
+
+There are two modes of operation (`resolvers`) for flagd providers; _in-process_ and _RPC_.
+Both modes have their advantages and disadvantages.
+For more information on the architectural implications of these different modes, see the [RPC vs In-Process Evaluation](/architecture/#rpc-vs-in-process-evaluation) page.
+
+## Configuration
+
+### Configuration options
+
+Most options can be defined in the constructor, or as environment variables, with constructor options having the highest
+precedence.
+
+Below are the supported configuration parameters (note that not all apply to both resolver modes):
+
+| Option name        | Environment variable name  | Explanation                                                                     | Type & Values            | Default                       | Compatible resolver |
```

Comment:
We should include `FLAGD_TARGET_URI` in this table.

https://github.com/open-feature/go-sdk-contrib/pull/587/files

Reply:
Actually it's already in here as `targetUri`/`FLAGD_GRPC_TARGET`.

Reply:
Updated to match what's now in Java and Go.